cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/PredefinedNoConfidenceDRep.hs

```diff
@@ -0,0 +1,457 @@
+foldEpochStateCommitteee hasCommittee anyNewEpochState = do
+  let committee = filterCommittee anyNewEpochState
+  put committee
+  if null committee == not hasCommittee
+  then return ConditionMet
+  else return ConditionNotMet
+
+filterCommittee :: AnyNewEpochState -> [(L.Credential L.ColdCommitteeRole L.StandardCrypto, EpochNo)]
+filterCommittee (AnyNewEpochState sbe newEpochState) =
+  caseShelleyToBabbageOrConwayEraOnwards
+    (const $ error "filterNoCommittee: Only conway era supported")
+    (const $ do
+      let rState = L.extractDRepPulsingState $ newEpochState ^. L.newEpochStateGovStateL . L.drepPulsingStateGovStateL
+          ensCommittee = rState ^. L.rsEnactStateL . L.ensCommitteeL
+      case ensCommittee of
+        SNothing -> []
+        SJust x -> Map.toList $ L.committeeMembers x
+    )
+    sbe
+
+updateConstitutionalCommittee
+  :: (MonadTest m, MonadIO m, H.MonadAssertion m, MonadCatch m, Foldable t)
+  => H.ExecConfig
+  -> EpochStateView
+  -> FilePath
+  -> FilePath
+  -> ConwayEraOnwards ConwayEra
+  -> FilePath
+  -> FilePath
+  -> PaymentKeyInfo
+  -> Maybe (String, Word32)
+  -> t (Int, String)
+  -> Integer
+  -> m (String, Word32)
+updateConstitutionalCommittee execConfig epochStateView configurationFile socketPath ceo work prefix
+                              wallet previousProposalInfo votes waitTillEpoch = do
+  baseDir <- H.createDirectoryIfMissing $ work </> prefix
+
+  let propVotes :: [(String, Int)]
+      propVotes = zip (concatMap (uncurry replicate) votes) [1..]
+  annotateShow propVotes
+
+  let coldVKeyFile = baseDir </> "cold-key.vkey"
+      coldSKeyFile = baseDir </> "cold-key.skey"
+
+  void $ H.execCli' execConfig
+      [ "conway", "governance", "committee", "key-gen-cold"
+      , "--cold-verification-key-file", coldVKeyFile
+      , "--cold-signing-key-file", coldSKeyFile
+      ]
+
+  coldKeyHash <- Text.unpack . Text.strip . Text.pack <$> H.execCli' execConfig
+    [ "conway", "governance", "committee", "key-hash"
+    , "--verification-key-file", coldVKeyFile
+    ]
+
+  thisProposal@(governanceActionTxId, governanceActionIndex) <-
+    makeUpdateConstitutionalCommitteeProposal execConfig epochStateView (File configurationFile) (File socketPath)
+                                              ceo baseDir "proposal" previousProposalInfo [coldKeyHash] wallet
+
+  voteChangeProposal execConfig epochStateView ceo baseDir "vote"
+                     governanceActionTxId governanceActionIndex propVotes (zip (repeat "yes") [1..3]) wallet
+
+  (EpochNo epochAfterProp) <- getCurrentEpochNo epochStateView
+  H.note_ $ "Epoch after \"" <> prefix <> "\" prop: " <> show epochAfterProp
+
+  committee <- waitTillCommittee True configurationFile socketPath waitTillEpoch
+  H.note_ $ show committee
+
+  return thisProposal
+
+makeUpdateConstitutionalCommitteeProposal
+  :: (H.MonadAssertion m, MonadTest m, MonadCatch m, MonadIO m, Foldable f)
```

Comment:
```suggestion
  :: (HasCallStack, H.MonadAssertion m, MonadTest m, MonadCatch m, MonadIO m, Foldable f)
```

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5809/commits/62a3cbbad2eaa4573630cc9b7aa40f74f1ae7e41

Reply:
Fixed in: https://github.com/IntersectMBO/cardano-node/pull/5809/commits/c69038ffc8865e0591f23aeb64dad34dc669bb58